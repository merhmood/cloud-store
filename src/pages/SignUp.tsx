/**
 * node_modules packages should always be addded to the section
 * denoted as node modules, while other imports should be added to 
 * the section denoted as app created, this helps in organization 
 * of import statements <3 
 */

//node_modules
import React from 'react'
//import validator from 'validator'

//app created
import Logo from '../components/Logo'
import BaseForm from '../components/BaseForm'

const SignUp = () => {
  return (
    <form>
      <Logo />
      <input type='text' placeholder='Username' />
      <BaseForm buttonText='Sign Up'/>
    </form>
  )
}

export default SignUp