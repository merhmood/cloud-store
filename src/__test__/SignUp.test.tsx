/**
 * node_modules packages should always be addded to the section
 * denoted as node modules, while other imports should be added to
 * the section denoted as app created, this helps in organization
 * of import statements <3
 */

//node_modules
import React from 'react';
import renderer from 'react-test-renderer'

//app created
import SignUp from '../pages/SignUp'

test('renders Sign up form', () => {
  const tree = renderer
    .create(<SignUp />)
    .toJSON();
  expect(tree).toMatchSnapshot()
});
