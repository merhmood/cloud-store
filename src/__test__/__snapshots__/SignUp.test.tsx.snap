// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Sign up form 1`] = `
<form>
  <p
    style={
      Object {
        "color": "black",
        "fontSize": "2rem",
        "fontWeight": "600",
        "marginBottom": "1em",
        "textAlign": "center",
      }
    }
  >
    Media
    <span
      style={
        Object {
          "color": "#dd2222",
        }
      }
    >
      Store
    </span>
  </p>
  <input
    placeholder="Username"
    type="text"
  />
  <input
    placeholder="Email"
    type="text"
  />
  <input
    placeholder="Password"
    type="password"
  />
  <label
    htmlFor="show password"
  >
    Show password
  </label>
  <input
    type="checkbox"
  />
  <button>
    Sign Up
  </button>
</form>
`;
